src/model/MintInfo.ts

```diff
@@ -0,0 +1,90 @@
+import { GetInfoResponse, MPPMethod, SwapMethod, WebSocketSupport } from './types';
+
+export class MintInfo {
+	private readonly mintInfo: GetInfoResponse;
+
+	constructor(info: GetInfoResponse) {
+		this.mintInfo = info;
+	}
+
+	isSupported(num: 4 | 5): { disabled: boolean; params: Array<SwapMethod> };
+	isSupported(num: 7 | 8 | 9 | 10 | 11 | 12 | 14): { supported: boolean };
+	isSupported(num: 17): { supported: boolean; params?: Array<WebSocketSupport> };
+	isSupported(num: 15): { supported: boolean; params?: Array<MPPMethod> };
+	isSupported(num: number) {
+		switch (num) {
+			case 4:
+			case 5: {
+				return this.checkMintMelt(num);
+			}
+			case 7:
+			case 8:
+			case 9:
+			case 10:
+			case 11:
+			case 12:
+			case 14: {
+				return this.checkGenericNut(num);
+			}
+			case 17: {
+				return this.checkNut17();
+			}
+			case 15: {
+				return this.checkNut15();
+			}
+		}
+	}
+	private checkGenericNut(num: 7 | 8 | 9 | 10 | 11 | 12 | 14) {
+		if (this.mintInfo.nuts[num]?.supported) {
+			return { supported: true };
+		}
+		return { supported: false };
+	}
+	private checkMintMelt(num: 4 | 5) {
+		const mintMeltInfo = this.mintInfo.nuts[num];
+		if (mintMeltInfo && mintMeltInfo.methods.length > 0 && !mintMeltInfo.disabled) {
+			return { disabled: false, params: mintMeltInfo.methods };
+		}
+		return { disabled: true, params: mintMeltInfo.methods };
+	}
+	private checkNut17() {
+		if (this.mintInfo.nuts['17'] && this.mintInfo.nuts[17].supported.length > 0) {
```

Comment:
Can all the keys just be numbers instead of nuts['17'] and nuts[17]? Also in `checkNut15`

Reply:
Yes, I will change all the accessors to numbers. However the type will keep the strings, as property names can not be numbers. Accessing using numbers only works because JS will coerce to strings automatically

Reply:
Or maybe I am misunderstanding, because I was referring to using `nuts[15]` instead of `nuts['15']`. Can you give an example?

Reply:
I think you got my point. I was referring to the accessors because in some places you used a string and some you used a number to access the value. I don't care if its string or number, but should be consistent by using one or the other.